src/mlpack/bindings/python/tests/test_python_binding.py

```diff
@@ -1336,5 +1336,37 @@ def testModelForceCopy(self):
     self.assertEqual(output2['model_bw_out'], 20.0)
     self.assertEqual(output3['model_bw_out'], 20.0)
 
+  def testCheckInputMatricesNaN(self):
+    """
+    Checks that an exception is thrown if the input matrix contains
+    NaN values.
+    """
+    x = np.random.rand(100, 5)
+    x[0][0] = np.nan
```

Comment:
Perhaps a good idea to set the value at a random position, so we don't test the first position all the time?